Feature/rnp 109

MichielVrins opened the pull request.

MichielVrins committed:
RNP-107: deregister fingerprint using generic deregister method

MichielVrins committed:
RNP-109: Rework ChangeAuthView and use generic registerAuthenticator

This commit makes the ChangeAuthView a lot easier to read by not passing
lots of state setters into a helper everywhere. This now only happens in
1 place which is properly named.
The generic helpers have now also been moved to a AuthenticationHelper
instead of being in FingerprintHelper and some of the redundant helper
functions have been removed.

MichielVrins committed:
RNP-109: Add await to async function call

Archifer commented on src/components/helpers/AuthenticatorHelper.ts:
```diff
@@ -0,0 +1,41 @@
+const getAllAuthenticators = async (
+  setRegisteredAuthenticators: (authenticators: Types.Authenticator[]) => void,
+  setAllAuthenticators: (authenticators: Types.Authenticator[]) => void,
+  setPreferred: (authenticator: Types.Authenticator) => void,
```
would change setPreferred to setPreferredAuthenticator like the two other params; but unsure if this would interfere with the const below that has the same name

Archifer commented on src/components/helpers/AuthenticatorHelper.ts:
```diff
@@ -0,0 +1,41 @@
+      profile.profileId,
+      preferred.id,
+    );
+    setMessage('The ' + preferred.name + ' is set');
```
so the messages would become for example; "The TouchId / fingerprint is set". I would maybe make it a bit more human readable in the form of: `The Authenticator "${preferred.name}" is now set as the default authentication method`

Also string templates might be nicer ;)

Archifer commented on src/components/screens/dashboard/components/ChangeAuthView.tsx:
```diff
@@ -4,17 +4,16 @@ import ContentContainer from './ContentContainer';
 import {AuthContext} from '../../../..//providers/auth.provider';
 import {useActionSheet} from '@expo/react-native-action-sheet';
 import Button from '../../../../components/general/Button';
+import OneginiSdk from 'onewelcome-react-native-sdk';
```
why is this oneginisdk; before in "fingerprinthelper" you were importing the onewelcomesdk; is there a difference?

Archifer commented on src/components/screens/dashboard/components/ChangeAuthView.tsx:
```diff
@@ -44,10 +43,23 @@ const ChangeAuthView: React.FC = () => {
+    getAllAuthenticators(
+      setRegisteredAuthenticators,
+      setAllAuthenticators,
+      setPreferred,
```
would change the name of setPreferred to indicate what you are setting

Archifer reviewed (commented):
see other specific comments

MichielVrins committed:
RNP-109: Refactor some variable names and message content

MichielVrins committed:
RNP-109: Make updating of the authenticators more robust

Previously if you would encounter an error the ui would not update
properly. With this commit we now update the authenticators when we
encounter en error during the enabling/disabling of fingerprint and
when setting the preferred authenticator. The retrieving of the
authenticators is now also properly error handled if any errors occur
during that process.

MichielVrins committed:
RNP-109: Add another errorcode for logout and refactor some method names

Previously we were not logging out on all of the no authenticated user
profile codes. This commit adds some.

MichielVrins committed:
RNP-109: Reference story in TODO message

PiotrKlis commented on src/components/helpers/AuthenticatorHelper.ts:
```diff
@@ -0,0 +1,43 @@
+) => {
+  const profile = await OneWelcomeSdk.getAuthenticatedUserProfile();
+  const registeredAuthenticators =
+    await OneWelcomeSdk.getRegisteredAuthenticators(profile.profileId);
```
I know that this wasn't introduced in this PR but you don't need `profile` prefix in member if object is called `profile`, could be just `profile.id`

PiotrKlis commented on src/components/helpers/AuthenticatorHelper.ts:
```diff
@@ -0,0 +1,43 @@
+  );
+
+  registeredAuthenticators.forEach(it => {
+    if (it.isPreferred) {
```
don't you have `find()` method or something similar at your disposal? https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/find.html

MichielVrins committed:
RNP-109: Fixup some typos

PiotrKlis commented on src/components/helpers/FingerprintHelper.ts:
```diff
@@ -1,97 +1,19 @@
     authenticator =>
-      authenticator.id == 'fingerprint' && authenticator.isRegistered,
+      authenticator.isRegistered &&
+      (authenticator.name.toUpperCase() === 'TOUCHID' ||
```
Isn't `TOUCHID` & `FINGERPRINT` a good candidate for class private enum? 🤔
Idk how it works here but in kotlin would be something like:

```
private enum class FINGERPRINT_TYPES(name: String) {
   FINGERPRINT("FINGERPRINT), 
   TOUCH_ID("TOUCHID")
}
```

MichielVrins replied:
I think we should actually expose these from the SDK as fingerprint/touchID/faceIdD should never change name/id as they come from the native sdks and not from the tokenserver, so let's add them in an enum in the SDK.

PiotrKlis commented on src/components/screens/dashboard/components/ChangeAuthView.tsx:
```diff
@@ -31,23 +31,40 @@ const pinRegisteredAuthenticator: Types.Authenticator = {
+
+  useEffect(() => {
+    updateAuthenticators();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
commented code?

MichielVrins replied:
Thats code to disable the linter to check for something specific which I don't want in this usecase as it adds unneeded complexity.

PiotrKlis reviewed (commented):
looks good, left few small remarks

MichielVrins committed:
RNP-109: Use .find instead of iterating over all elements

MichielVrins committed:
RNP-109: Use .id for Userprofile objects instead of .profileId

MichielVrins committed:
RNP-109: Use enum or biometrics and rename fingerprint->biometric

PiotrKlis commented on src/components/helpers/BiometricHelper.ts:
```diff
@@ -0,0 +1,20 @@
+import OneWelcomeSdk, {Types} from 'onewelcome-react-native-sdk';
+
+const isBiometricAuthenticatorRegistered = async (
+  returnEnable: (enabled: boolean) => void,
```
This parameter name is wrong, I bet you can think of something better 😅

MichielVrins committed:
RNP-109: Change signature of isBiometricAuthenticatorRegistered

Previously we were requiring a callback which we would call with the
'enable' value. Now we are just returning a Promise<boolean> instead.

MichielVrins committed:
RNP-109: Update sdk version
